templates/templates/generic-component.yaml

```diff
@@ -0,0 +1,79 @@
+# DO NOT EDIT. THIS FILE IS GENERATED
+
+{{- /*
+   This template renders jobs for a component using generic buildpack with make and docker
+
+   Required values:
+     - repository - repository address (e.g. github.com/kyma-project/kyma
+     - path - address of the component within the repository
+     - pushRepository - the suffix of the preset-docker-push- label
+     - bootstrapTag - tag of the bootstrap image
+     - ReleaseBranchPattern - pattern for release branch prefix (default: ^release-<supported-releases>-<component-dir-name>$)
+     - runIfChanged - a list of regexps to watch in addition to path (default: [])
+     - resources.memory - memory given to job container (default: 1.5Gi)
+     - resources.cpu - cpu given to job container (default: 0.8)
+  */ -}}
+
+{{ $repositoryName := index (.Values.repository | splitList "/") 2 -}}
+{{- $additionalRunIfChanged := .Values.additionalRunIfChanged | default (list) }}
+{{- $runIfChanged := prepend $additionalRunIfChanged (print "^" .Values.path "/") | join "|" }}
+{{- $componentName := .Values.path | splitList "/" | last }}
+{{- $isNotTestInfra := ne .Values.repository "github.com/kyma-project/test-infra"}}
+{{- $hasReleaseBounds := (not .Values.since) and (not .Values.until) }}
+{{- $releases := matchingReleases .Global.releases .Values.since .Values.until }}
+{{- $releasesPattern := $releases | join "|" | replace "." "\\." }}
+{{- $releaseBranchReleasesPattern := $hasReleaseBounds | ternary ".*" (print "("  $releasesPattern ")") }}
+{{- $defaultReleaseBranchPattern := (print "^release-" $releaseBranchReleasesPattern "-" $componentName "$") }}
+{{- $defaultReleaseBranchPatternOrEmpty := ne (len $releases) 0 | ternary $defaultReleaseBranchPattern "" }}
+{{- $releaseBranchPattern := .Values.ReleaseBranchPattern | default $defaultReleaseBranchPatternOrEmpty }}
+
+{{- if $isNotTestInfra }}
+test_infra_ref: &test_infra_ref
+  org: kyma-project
+  repo: test-infra
+  path_alias: github.com/kyma-project/test-infra
+{{- end }}
+
+job_template: &job_template
+  decorate: true
+  path_alias: github.com/kyma-project/kyma
+  max_concurrency: 10
+  labels:
+    preset-dind-enabled: "true"
+    preset-sa-gcr-push: "true"
+    preset-docker-push-repository-{{ .Values.pushRepository }}: "true"
```

Comment:
Can we change it to something like this? 
```
{{- if {{ .Values.pushRepository }} }}
    preset-docker-push-repository: "true"
{{- else }}
   preset-docker-push-repository-{{ .Values.pushRepository }}: "true"
{{- end }}
```
or remove the label without repo value.

Reply:
I'm a fan of the second approach.

Reply:
I removed the preset-docker-push-repository and replaced it with preset-docker-push-repository-kyma. To align with others I also renamed global repository